udf_utils.go

```diff
@@ -32,12 +31,12 @@ func getPinned[T any](handle unsafe.Pointer) T {
 	return h.Value().(pinnedValue[T]).value
 }
 
-// Set error helpers
+// Set error helpers.
 
 func setBindError(info C.duckdb_bind_info, msg string) {
 	err := C.CString(msg)
-	defer C.duckdb_free(unsafe.Pointer(err))
 	C.duckdb_bind_set_error(info, err)
+	C.duckdb_free(unsafe.Pointer(err))
```

Comment:
It is common in go to always use defer even if it looks stupid, since it means that the same pattern can always be applied, and you don't have to think about it later in the code. It is also more resistant to code-changes, as whatever you do it will always be cleaned up correctly. But I saw this this is also done this way in the rest of the code, so if you don't care just LMK and I will merge it

Reply:
In loops, my IDE gives me warning when using `defer`... But I am comparatively new to Go, so I'll follow your suggestion. :) I'll update all non-loop occurrences. 👍

Reply:
Defer executes it at the end of the function, so within loops it may execute them later than you expect. When I was more active in the go community this was considered a design mistake, but not much can be done about it now.
Your IDE probably complains because using defer within a loop is a bit confusing since it doesn't call them at the end of each loop iteration, which is fair.

Reply:
Yeah, that's probably the reason my IDE complains there.
It's great to have you active on the project, though, as you have more Go experience than I do. So, your input is much appreciated. I updated the files. 👍